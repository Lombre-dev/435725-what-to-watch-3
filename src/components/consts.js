export const GENRES = [
  `Comedy`,
  `Crime`,
  `Documentary`,
  `Drama`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thriller`,
];

export const RATING_SCORE_LIMIT = 10;
export const RATING_LEVELS = [
  `Bad`,
  `Normal`,
  `Good`,
  `Very good`,
  `Awesome`,
];
