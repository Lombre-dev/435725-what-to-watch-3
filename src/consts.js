export const ALL_GENRE = `All genres`;
const COMEDY_GENRE = `Comedy`;
const CRIME_GENRE = `Crime`;
const DOCUMENTARY_GENRE = `Documentary`;
const DRAMA_GENRE = `Drama`;
const HORROR_GENRE = `Horror`;
const KIDS_AND_FAMILY_GENRE = `Kids & Family`;
const ROMANCE_GENRE = `Romance`;
const SCI_FI_GENRE = `Sci-Fi`;
const THRILLER_GENRE = `Thriller`;

export const AppPages = {
  MAIN: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  MOVIES: `/films`,
  PLAYER: `/player`,
};

export const GENRES = [
  COMEDY_GENRE,
  CRIME_GENRE,
  DOCUMENTARY_GENRE,
  DRAMA_GENRE,
  HORROR_GENRE,
  KIDS_AND_FAMILY_GENRE,
  ROMANCE_GENRE,
  SCI_FI_GENRE,
  THRILLER_GENRE,
];

export const GenreLabels = {
  [COMEDY_GENRE]: `Comedies`,
  [DRAMA_GENRE]: `Dramas`,
  [THRILLER_GENRE]: `Thrillers`,
};

export const MORE_LIKE_THIS_MOVIES_LIMIT = 4;
export const CATALOG_MOVIES_PER_PAGE_LIMIT = 8;

export const SMALL_MOVIE_CARD_PREVIEW_DELAY = 1000;

export const MOVIE_INFO_TABS = [
  `Overview`,
  `Details`,
  `Reviews`,
];

export const PlayerState = {
  INITED: `inited`,
  LOADING: `loading`,
  PLAYING: `playing`,
  PAUSED: `paused`,
  ENDED: `ended`,
};

export const SERVER_URL = `https://htmlacademy-react-3.appspot.com/wtw`;
export const SERVER_RESPONSE_TIMEOUT = 5000;
export const SERVER_USE_COOKIES = true;
export const ServerErrors = {
  NOT_FOUND: 404,
  UNAUTHORIZED: 401,
};
