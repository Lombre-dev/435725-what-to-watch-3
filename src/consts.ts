export const ALL_GENRE = `All genres`;
const COMEDY_GENRE = `Comedy`;
const CRIME_GENRE = `Crime`;
const DOCUMENTARY_GENRE = `Documentary`;
const DRAMA_GENRE = `Drama`;
const HORROR_GENRE = `Horror`;
const KIDS_AND_FAMILY_GENRE = `Kids & Family`;
const ROMANCE_GENRE = `Romance`;
const SCI_FI_GENRE = `Sci-Fi`;
const THRILLER_GENRE = `Thriller`;

export enum AppPages {
  MAIN = `/`,
  LOGIN = `/login`,
  MY_LIST = `/mylist`,
  MOVIES = `/films`,
  PLAYER = `/player`,
}

export const GENRES = [
  COMEDY_GENRE,
  CRIME_GENRE,
  DOCUMENTARY_GENRE,
  DRAMA_GENRE,
  HORROR_GENRE,
  KIDS_AND_FAMILY_GENRE,
  ROMANCE_GENRE,
  SCI_FI_GENRE,
  THRILLER_GENRE,
];
export const GENRES_CATALOG_LIMIT = 9;

export const GenreLabels = {
  [COMEDY_GENRE]: `Comedies`,
  [DRAMA_GENRE]: `Dramas`,
  [THRILLER_GENRE]: `Thrillers`,
};

export const MORE_LIKE_THIS_MOVIES_LIMIT = 4;
export const CATALOG_MOVIES_PER_PAGE_LIMIT = 8;

export const SMALL_MOVIE_CARD_PREVIEW_DELAY = 1000;

export const MOVIE_INFO_TABS = [
  `Overview`,
  `Details`,
  `Reviews`,
];

export enum PlayerState {
  INITED = `inited`,
  LOADING = `loading`,
  PLAYING = `playing`,
  PAUSED = `paused`,
  ENDED = `ended`,
}

export const ServerConfig = {
  URL: `https://htmlacademy-react-3.appspot.com/wtw`,
  RESPONSE_TIMEOUT: 5000,
  USE_COOKIES: true,
};
export enum ServerErrors {
  NOT_FOUND = 404,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
}

export enum AuthorizationErrorCode {
  INCORRECT_LOGIN = 1,
  AUTHORIZATION_FAIL = 2,
}

export const AuthorizationErrorMessage = {
  [AuthorizationErrorCode.INCORRECT_LOGIN]: `Please enter a valid email address`,
  [AuthorizationErrorCode.AUTHORIZATION_FAIL]: `We canâ€™t recognize this email and password combination. Please try again.`,
};

export enum AuthorizationStatus {
  NO_AUTH = `unauthorized`,
  AUTH = `authorized`,
}

export const REVIEW_RATING_STARS_LIMIT = 5;
export const REVIEW_COMMENT_MIN_LENGTH = 50;
export const REVIEW_COMMENT_MAX_LENGTH = 400;

export enum LoadingDataStatus {
  LOADING = `loading`,
  READY = `ready`,
  ERROR = `error`,
}
