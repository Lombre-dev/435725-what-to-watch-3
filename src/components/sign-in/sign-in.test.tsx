import * as React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {AuthorizationStatus} from '../../consts';
import SignIn from './sign-in';

const mockStore = configureStore([]);
const HANDLE_EVENT = function () {};

describe(`<SignIn />`, () => {

  it(`render should be match markup`, () => {

    const store = mockStore({
      user: {
        id: -1,
        name: undefined,
        email: undefined,
        avatar: undefined,
        authError: undefined,
        authStatus: AuthorizationStatus.NO_AUTH,
      }
    });

    const result = renderer
      .create(<Provider store={store}>
        <BrowserRouter>
          <SignIn
            onSubmit={HANDLE_EVENT}
          />
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
