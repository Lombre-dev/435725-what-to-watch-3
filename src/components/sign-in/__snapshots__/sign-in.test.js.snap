// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> render should be match markup 1`] = `
<div
  className="sign-in user-page__content"
>
  <form
    action="#"
    className="sign-in__form"
    onSubmit={[Function]}
  >
    <div
      className="sign-in__fields"
    >
      <div
        className="sign-in__field"
      >
        <input
          className="sign-in__input"
          defaultValue="test@test.com"
          id="user-email"
          name="user-email"
          placeholder="Email address"
          type="email"
        />
        <label
          className="sign-in__label visually-hidden"
          htmlFor="user-email"
        />
      </div>
      <div
        className="sign-in__field"
      >
        <input
          className="sign-in__input"
          defaultValue="1"
          id="user-password"
          name="user-password"
          placeholder="Password"
          type="password"
        />
        <label
          className="sign-in__label visually-hidden"
          htmlFor="user-password"
        >
          1
        </label>
      </div>
    </div>
    <div
      className="sign-in__submit"
    >
      <button
        className="sign-in__btn"
        type="submit"
      >
        Sign in
      </button>
    </div>
  </form>
</div>
`;
